test/unit/puppetlabs/puppetserver/certificate_authority_test.clj

```
@@ -1204,11 +1204,26 @@
                                     :value    "true"}
                                    {:oid      "2.5.29.17"
                                     :critical false
-                                    :value    {:dns-name ["subject"
-                                                          "onefish"
-                                                          "twofish"]}}])]
+                                    :value    {:dns-name ["onefish"
+                                                          "twofish"
+                                                          "threefish"
+                                                          "fourfish"
+                                                          "subject"]}}])]
         (is (= (set exts) (set exts-expected)))))
 
+    (testing "correct subject alt name extensions are created for a master"
+      (let [dns-alt-names "onefish,twofish,DNS:threefish,IP:fourfish,fivefish,IP:sixfish"
+            exts-expected {:oid      "2.5.29.17"
+                            :critical false
+                            :value    {:dns-name ["onefish"
+                                                  "twofish"
+                                                  "threefish"
+                                                  "fivefish"
+                                                  "subject"]
+                                       :ip       ["fourfish"
```

I would expect this to fail with an invalid ip.